hack/generator/pkg/astmodel/allof_type.go

```diff
@@ -134,15 +134,15 @@ func (allOf AllOfType) RequiredPackageReferences() *PackageReferenceSet {
 // Equals returns true if the other Type is a AllOf that contains
 // the same set of types
 func (allOf *AllOfType) Equals(t Type) bool {
+	if allOf == t {
+		return true // short-circuit
+	}
+
```

Comment:
Won't this always fail because (here) they're different types? I thought the type conversion was needed first.

Reply:
It will do a comparison with what is inside the interface, so equivalent to doing a pointer comparison.